scripts/docs_link_checker.py

```diff
@@ -0,0 +1,306 @@
+import glob
+import logging
+import os
+import re
+from typing import List
+
+import click
+import requests
+
+logger = logging.getLogger()
+logger.addHandler(logging.StreamHandler())
+logger.setLevel(logging.INFO)
+
+
+class LinkReport:
+    def __init__(self, link: str, file: str, message: str):
+        self.link = link
+        self.file = file
+        self.message = message
+
+    def __str__(self):
+        return f"{self.message}: File: {self.file}, Link: {self.link}"
+
+
+class LinkChecker:
+    def __init__(
+        self, docs_path: str, docs_root: str, site_prefix: str, skip_external: False
+    ):
```

Comment:
could we annotate `skip_external` as a `bool`? The default value is meant to be `False` here right?

Reply:
Done!